src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/summaryjavadoc/InputSummaryJavadocInlineReturn.java

```diff
@@ -0,0 +1,45 @@
+/*
+SummaryJavadoc
+violateExecutionOnNonTightHtml = (default)false
+forbiddenSummaryFragments = (default)^$
+period = (default).
+
+*/
+
+package com.puppycrawl.tools.checkstyle.checks.javadoc.summaryjavadoc;
+
+class InputSummaryJavadocInlineReturn {
```

Comment:
We need some more examples.
What about if there is text before the `{@return}`.
What about if there is text after the `{@return}`.
What about if the content of `{@return}` spans multiple lines?
What about if a void method uses the `{@return}`? Does the javadoc flag an error on this? Do we?

Reply:
I will check the first 3, for the last one `javadoc` produces a hard error:
```
SummaryJavadocNewInlineReturn.java:3: error: invalid use of @return
     * {@return a foo that is {@code 0}}
       ^
```
so I think checking for that in checkstyle is redundant.

Reply:
> What about if there is text before the `{@return}`.

`javadoc` produces a warning. Should I add a warning to checkstyle as well? Should I add a test that this is accepted by checkstyle?
```
SummaryJavadocNewInlineReturn.java:5: warning: {@return} not at beginning of description
     * Starting text, {@return a foo that is {@code 0}} extra text afterwards.
                      ^
```

> What about if there is text after the `{@return}`.

This is normal and allowed, I will add a test that it does not produce a violation.

> What about if the content of `{@return}` spans multiple lines?

Using a normal newline, this produces output like the following:

![javadoc render](https://url.octyl.net/xyj2tan-uaoyf47t7mfz)

Using `<br>`, this produces output like the following:

![javadoc render](https://url.octyl.net/4cla-136hl5782mlzlbz)

I do not know if checkstyle should produce a message for either of these. Please inform me as to what is appropriate.

Reply:
> for the last one javadoc produces a hard error:

Is there any issue adding this? We specifically have another where it also produces an error by the javadoc.
https://github.com/checkstyle/checkstyle/pull/10772#discussion_r814915451

> I will add a test

Ok.

> javadoc produces a warning. Should I add a warning to checkstyle as well?

@romani please comment on scope of issue.

> Please inform me as to what is appropriate.

Please update your post to include the raw code of the javadoc, so I can correlate that to the javadoc printouts you provided.

Reply:
We should not do what is already done by javadoc

Reply:
> Please update your post to include the raw code of the javadoc, so I can correlate that to the javadoc printouts you provided.

```java
/**
 */
public class SummaryJavadocNewInlineReturn {
    /**
     * {@return a foo that is {@code 0}<br>
     * this takes a lot<br>
     * of lines to explain<br>
     * so it is quite long}
     */
    public int foo() {
    }
}
```
Remove the `<br>`s for the first output.

Reply:
> I do not know if checkstyle should produce a message for either of these.

As far as I can tell, the javadoc accepts everything as the summary. Checkstyle should do the same.

Were you asking for advice on something else?

Reply:
No, thank you.

Reply:
This is all done now.